### ISSUE TITLE
Set explicit version of SecretStorage

### ISSUE THREAD

**Issue by jafreck** [jaraco/keyring#321]
Version 3.0.0 of secretstorage just released and has breaking changes. This dependency does not have an explicit version in setup.py.

https://github.com/jaraco/keyring/blob/1e25fde80b5e45cc25fe3efdf4edded27706578b/setup.py#L62

**Comment by lmazuel** [comment]
Indeed, got broken by that as well. Best fix would be IMHO to release a keyring 12.0.2 that specifies "requires" as `secretstorage<3.0.0`.

**Comment by jladdjr** [comment]
Issue description mentions secretservice, but dep mentioned is secretstorage.

**Comment by jafreck** [comment]
@jladdjr Title should have read secretstorage, fixed the typo.

**Comment by mitya57** [comment]
This will be fixed as part of #320. For Python 3.5+, we **do** want SecretStorage 3.x.

**Comment by jaraco** [comment]
I've released keyring 12.0.2 pinned to SecretStorage 2 and 12.1 allowing SecretStorage 3.

**Comment by lmazuel** [comment]
@jaraco This means that starting keyring 12.1, keyring is not compatible with Python 2.7 and 3.4 on Linux (since you cannot install the underlying dependencies). Dropping some support of Python should be at least a major version IMHO and a 13.x series. Or are you saying that you will decide what version of secretstorage to install depending of the running Python?
My two cents.

**Comment by mitya57** [comment]
@lmazuel Why? Just continue using the old version of SecretStorage, and you will be fine.

**Comment by lmazuel** [comment]
@mitya57 "SecretStorage" is a dependency of "keyring". I'm not supposed to care about your backend, and then care about the version of SecretStorage I should install. "pip install keyring" should just work.
But if you don't pin SecretStorage in keyring 12.1, this means "pip install keyring==12.1" on Python 2.7 will fail, because it will download the sdist of SecretStorage, and then fail to install it. If you can't install "keyring" on Python 2.7, whatever the underlying cause, this means you don't support Python 2.7 anymore :). If you don't support Python 2.7 anymore, I think it's worth a major version bump ;)

Another solution is to write something like:
```python
    extras_require={
        ":python_version<='3.4'": ['secretstorage<3.0.0'],
        ":python_version>='3.5'": ['secretstorage'],
    }
```
But then you need to be sure keyring stays compatible with latest SecretStorage and secretstorage 2.x at the same time.

**Comment by jaraco** [comment]
@lmazuel Isn't that what [keyring currently does](https://github.com/jaraco/keyring/blob/12.1.0/setup.py#L61-L67)?

**Comment by lmazuel** [comment]
@jaraco ho, that's cool, we're on the same page, I misunderstood you sentence:
> I've released keyring 12.0.2 pinned to SecretStorage 2 and 12.1 allowing SecretStorage 3.

Sorry about the noise here :). Thanks for the awesome job :)
